sphinx/transforms/__init__.py

```diff
@@ -327,3 +329,33 @@ def visit_pending_xref(self, node):
     def visit_image(self, node):
         # type: (nodes.Node) -> None
         raise nodes.SkipNode
+
+
+class SphinxSmartQuotes(DocutilsSmartQuotes):  # NOQA
+    """
+    Customized SmartQuotes to avoid transform for some extra node types.
+    """
+    def get_tokens(self, txtnodes):
+        # A generator that yields ``(texttype, nodetext)`` tuples for a list
+        # of "Text" nodes (interface to ``smartquotes.educate_tokens()``).
+
+        texttype = {True: 'literal',  # "literal" text is not changed:
+                    False: 'plain'}
+        for txtnode in txtnodes:
+            nodetype = texttype[isinstance(txtnode.parent,
+                                           (nodes.literal,
+                                            nodes.literal_block,
+                                            addnodes.literal_emphasis,
+                                            addnodes.literal_strong,
+                                            addnodes.desc_signature,
+                                            addnodes.productionlist,
+                                            addnodes.desc_optional,
+                                            addnodes.desc_name,
```

Comment:
Oh, interesting -- so ***all*** uses of `addnodes.desc_name` should be rendered as literal? I wouldn't have anticipated that.